File: 0554-brick-wall/0554-brick-wall.py
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        
        hashmap = {0:0}
        
        for row in wall:
            gapIndex = 0
            for brick in row[:-1]:
                gapIndex += brick
                hashmap[gapIndex] = 1 + hashmap.get(gapIndex, 0)
        
        maxgapNumber = max(hashmap.values())
        return len(wall) - maxgapNumber

File: 0554-brick-wall/0554-brick-wall.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        int rowsize = size(wall), maxgapNumber = 0;
        unordered_map<int, int> hashmap;
        
        for(auto& row : wall){
            int gapIndex = 0;
            for(int i = 0; i < size(row) - 1; i++){
                gapIndex += row[i];
                hashmap[gapIndex] ++;
                maxgapNumber = max(maxgapNumber, hashmap[gapIndex]);

            }
        }
        return rowsize  - maxgapNumber;
    }
};